src/SelfService/Application/MembershipApplicationService.cs

```diff
@@ -36,10 +38,20 @@ IMembershipApplicationDomainService membershipApplicationDomainService
         _systemTime = systemTime;
         _membershipQuery = membershipQuery;
         _membershipApplicationDomainService = membershipApplicationDomainService;
+        _invitationRepository = invitationRepository;
     }
 
     private async Task CreateAndAddMembership(CapabilityId capabilityId, UserId userId)
     {
+        // Note: requires [TransactionalBoundary], which should be wrapped elsewhere
+        var invitation = await _invitationRepository.FindByPredicate(
+            x => x.Invitee == userId && x.TargetId == capabilityId
+        );
+        if (invitation != null)
+        {
+            invitation.Cancel();
+        }
```

Comment:
Here we are looking for a single invitation, but in InvitationApplicationService we are using GetAllWithPredicate

Reply:
Good catch; Of course it should be all -- and also only active ones.

Again we should never have more than one, but to be on the safe side